### PR TITLE
Feature show meals of user in dashboard + additional information -  cu b0u3pd

### PR DESCRIPTION
**Pull request by MalteBoehm**
While working on this feature I had to change a lot about the API requests. The data I got from there was not clean enough.

### CODE REVIEW

**Comment by MalteBoehm** [comment]
Task linked: [CU-b0u3pd (Backend + Frontend) Mahlzeit soll im Frontend angezeigt werden](https://app.clickup.com/t/b0u3pd)

File: frontend/src/screens/AddMealScreen/AddMealList.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  const [checked, setChecked] = useState(false);
+
+  const keyExtractor = (item, index) => index.toString();
+  const renderItem = ({ item }) => {
+    let isChecked = false;
+
+    const handleChecked = () => {
+      if (!addMealListOfProducts.includes(item)) {
+        addMealListOfProducts.push(item);
```
Comment (jamarob) [inline frontend/src/screens/AddMealScreen/AddMealList.js:23]: Be careful with directly modifying props or context variables, might ot work as intended

File: frontend/src/screens/AddMealScreen/AddMealList.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import AddMealSuche from "./AddMealSuche";
+
+
+export default function AddMealList() {
+  const { liveSearchData, addMealListOfProducts, setAddMealListOfProducts } = useContext(LiveSearchContext);
+
+  const [checked, setChecked] = useState(false);
+
+  const keyExtractor = (item, index) => index.toString();
+  const renderItem = ({ item }) => {
```
Comment (jamarob) [inline frontend/src/screens/AddMealScreen/AddMealList.js:18]: Better move this to its own component file

File: frontend/src/screens/AddMealScreen/AddMealSuche.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { useContext } from "react";
+import { Row } from "react-native-easy-grid";
+import { SearchBar } from "react-native-elements";
+import SearchInputContext from "../../context/SearchInputContext";
+
+
+export default function AddMealSuche() {
```
Comment (jamarob) [inline frontend/src/screens/AddMealScreen/AddMealSuche.js:7]:
```suggestion
export default function AddMealSearch() {
```

File: frontend/src/screens/DashboardScreen/DashboardList.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import DashboardMealHeader from "./components/DashboardMealHeader";
+import DashboardMealAllergens from "./components/DashboardMealAllergens";
+import DashboardMealSideEffects from "./components/DashboardMealSideEffects";
+import DashboardContext from "../../context/DashboardContext";
+
+export default function DashboardList({ navigation }) {
+    const { todaysBreakfast, todaysMeal, todaysDinner, todaysSnack } = useContext(DashboardContext);
+
+
+    const listItemsToMap = [
```
Comment (jamarob) [inline frontend/src/screens/DashboardScreen/DashboardList.js:17]: Move constant data out of the component, it doesn't have to be redefined on every render.

File: frontend/src/screens/DashboardScreen/DashboardList.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+
+    return (
+      <Row size={2}>
+          <ScrollView>
+              <Grid style={GridListStyled.container}>
+                  {listItemsToMap.map(item => {
+
+                      const products = item.mapObject.map(meal => meal.products?.map(product => product.product_name)).flat();
+                      const allergens = item.mapObject.map(meal => meal.allergens?.map(allergen => allergen.names)).flat();
```
Comment (jamarob) [inline frontend/src/screens/DashboardScreen/DashboardList.js:52]: please simplify your logic ;)

File: frontend/src/screens/DashboardScreen/DashboardList.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                      const hasSideEffects = item.mapObject.hasSideEffect;
+                      const hasAllergens = item.mapObject.allergens;
+                      const checkSideEffectsArray = item.mapObject.sideEffects;
```
Comment (jamarob) [inline frontend/src/screens/DashboardScreen/DashboardList.js:55]: Be consistent in your variable names, whats the difference between `item.mapObject.hasSideEffect` and `item.mapObject.sideEffects` if the latter is indeed an array `hasSideEffects` can be derived from the arrays length 🤔

**Review by jamarob — COMMENTED** [review]
That was a long one :) please fix your formatting and try to be clearer with your naming. You do a lot of for loops that stem from your shape of your datastructure.. maybe streamline that if possible.

**Pull request merged**